### PR TITLE
Only Load Cases/Users with GPS Data on Map

### PR DESCRIPTION
**Pull request by zandre-eng**
## Product Description

The buttons for cases/users missing GPS data has been removed:
missing-gps-btns-removed

## Technical Summary

Link to ticket [here](https://dimagi-dev.atlassian.net/browse/SC-3257).

The queries for retrieving cases/users for the "Case Management" page have been tweaked to filter out cases and users that do not contain GPS data. This was done due to the fact that loading in these cases/users provides no real value to the map, and the user can rather go to the "Manage GPS Data" page to manage cases/users that are missing GPS data.

## Feature Flag

`GEOSPATIAL`

### Safety story

- Local testing.
- Automated tests exist.

### Automated test coverage

There is test coverage for the relevant views to ensure that the correct filtering is being applied and that valid data gets returned.

### QA Plan

No QA planned.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by SmittieC — APPROVED** [review]

File: corehq/apps/geospatial/static/geospatial/js/geospatial_map.js [hunk 1]
```diff
@@ -339,16 +337,7 @@ hqDefine("geospatial/js/geospatial_map", [
                     self.hasFiltersChanged(false);
 
                     // TODO: There is a lot of indexing happening here. This should be replaced with a mapping to make reading it more explicit
```
Comment (kaapstorm) [inline corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:339]: @zandre-eng is this comment still valid? It looks like you've avoided doing what the comment was about ... or have I misunderstood the comment?

Comment (zandre-eng) [reply corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:339]: Good question. I do have another refactor PR which is why I avoided addressing the comment here. Looking through the code however, this function has been simplified a lot, so there really isn't that much indexing happening anymore. I'm thining we can probably remove this comment, unless you have any objections @kaapstorm?

Comment (kaapstorm) [reply corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:339]: I don't have any objections to removing the comment. :)

Comment (zandre-eng) [reply corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:339]: I've made a [separate PR](https://github.com/dimagi/commcare-hq/pull/33996) to remove this comment. Thanks for raising this!

**Pull request merged**